server/core/storelimit/store_limit.go

```diff
@@ -64,45 +91,45 @@ func (t Type) String() string {
 	return ""
 }
 
-// StoreLimit limits the operators of a store
-type StoreLimit struct {
-	limiter         *ratelimit.RateLimiter
-	regionInfluence int64
-	ratePerSec      float64
+// limit  the operators of a store
+type limit struct {
+	limiter    *ratelimit.RateLimiter
+	ratePerSec float64
 }
 
-// NewStoreLimit returns a StoreLimit object
-func NewStoreLimit(ratePerSec float64, regionInfluence int64) *StoreLimit {
-	capacity := regionInfluence
+func (l *limit) Reset(ratePerSec float64) {
+	if l.ratePerSec == ratePerSec {
+		return
+	}
+	capacity := int64(influence)
 	rate := ratePerSec
 	// unlimited
 	if rate >= Unlimited {
 		capacity = int64(Unlimited)
 	} else if ratePerSec > 1 {
-		capacity = int64(ratePerSec * float64(regionInfluence))
-		ratePerSec *= float64(regionInfluence)
+		capacity = int64(ratePerSec * float64(influence))
+		ratePerSec *= float64(influence)
 	} else {
-		ratePerSec *= float64(regionInfluence)
-	}
-	return &StoreLimit{
-		limiter:         ratelimit.NewRateLimiter(ratePerSec, int(capacity)),
-		regionInfluence: regionInfluence,
-		ratePerSec:      rate,
+		ratePerSec *= float64(influence)
 	}
+	l.limiter = ratelimit.NewRateLimiter(ratePerSec, int(capacity))
+	l.ratePerSec = rate
 }
 
 // Available returns the number of available tokens
-func (l *StoreLimit) Available(n int64) bool {
+// It returns true if the rate per second is zero.
+func (l *limit) Available(n int64) bool {
+	if l == nil || l.ratePerSec == 0 {
```

Comment:
When will it become nil?

Reply:
yes, it will not be nil even if the rate per second is zero, I will remove this condition.